src/lib/operators/jit_operator/operators/jit_compute.cpp

```diff
@@ -0,0 +1,213 @@
+#include "jit_compute.hpp"
+
+#include "../jit_operations.hpp"
+#include "constant_mappings.hpp"
+
+namespace opossum {
+
+JitExpression::JitExpression(const JitTupleValue& tuple_value)
+    : _expression_type{ExpressionType::Column}, _result_value{tuple_value} {}
+
+JitExpression::JitExpression(const JitExpression::Ptr& child, const ExpressionType expression_type,
+                             const size_t result_tuple_index)
+    : _left_child{child},
+      _expression_type{expression_type},
+      _result_value{JitTupleValue(_compute_result_type(), result_tuple_index)} {}
+
+JitExpression::JitExpression(const JitExpression::Ptr& left_child, const ExpressionType expression_type,
+                             const JitExpression::Ptr& right_child, const size_t result_tuple_index)
+    : _left_child{left_child},
+      _right_child{right_child},
+      _expression_type{expression_type},
+      _result_value{JitTupleValue(_compute_result_type(), result_tuple_index)} {}
+
+std::string JitExpression::to_string() const {
+  if (_expression_type == ExpressionType::Column) {
+    return "x" + std::to_string(_result_value.tuple_index());
+  }
+
+  const auto left = _left_child->to_string() + " ";
+  const auto right = _right_child ? _right_child->to_string() + " " : "";
+  return "(" + left + expression_type_to_string.at(_expression_type) + " " + right + ")";
+}
+
+void JitExpression::compute(JitRuntimeContext& ctx) const {
+  // We are dealing with an already computed value here, so there is nothing to do.
+  if (_expression_type == ExpressionType::Column) {
+    return;
+  }
+
+  auto left_value = _left_child->result().materialize(ctx);
+  auto result_value = _result_value.materialize(ctx);
+  _left_child->compute(ctx);
```

Comment:
The "materialize" / "compute" part is confusing to me. Could you clarify this?

Reply:
It was confusing. That is why it's gone now.